### ISSUE TITLE
Tileset/Blockset Editor

### ISSUE THREAD

**Issue by RevoSucks** [yenatch/pretmap#64]
This issue should have been made already.

**Comment by qwertzu33** [comment]
When selecting a tile/block, it would be great to have an option to move all tiles/blocks below down or to move all tiles/blocks below up. I made a small video showcasing what i want using excel. Imagine "a" stands for a place where a tile/block is. https://streamable.com/md853
It should also have a warning if a column is full and moving the tiles/blocks down would remove the bottom tiles/blocks.

**Comment by cynderquil** [comment]
Being able to select and edit multiple blocks at a time would be a useful feature. For example, if you would like to add a large building from the tileset to the blockset, having this would reduce the time it would take to port the tiles over.

A blockset organizer would allow you to move around blocks within a selected blockset whilst also allowing you to change the amount of blocks.

**Comment by lilbluedemon** [comment]
The ability to edit the tileset's palettes in the tileset editor would be cool, like A-Map does~

**Comment by SBird1337** [comment]
Bear with me for this one:

What would be insanely useful for content creators would be a variant of the following wizardish functionality:

1. User selects an image which represents a finished Blockset
2. Wizard reduces the blockset, removing double tiles, indexing the image using 6 BG Palettes etc. kinda like GRIT does.
3. Wizard automatically creates a primary tileset and blockset which the user can use to map (Probably not taking behavior etc. into account, but he can do that manually later

Second:

1. User selects an image which represents a finished secondary blockset
2. User selects a primary blockset he wants to use that secondary set with
3. Wizard reduces the blockset again, this time taking into account tiles to be reused from the primary set
4. Wizard automagically creates a secondary tileset and blockset.

You'd probably have to take into account that there are 2 layers, so the user would have to provide those independently from each other, but it would make things insanely easier if you could at least sort of skip the process of mapping out your blockset all the time.

**Comment by huderlem** [comment]
@SBird1337 I had the exact same idea.  I am planning on prototyping that first to see if it's feasible (I think it is).  Instead of providing 2 layers, the user would provide 3 layers, since that's how the metatiles are actually represented in-game.  Layer 1 is drawn on top of the NPC and player sprites, while Layers 2 and 3 are drawn underneath the sprites.  Providing 3 layers eliminates the need for the user to retroactively assign the "background byte" for each metatile (a tedious and error-prone process).

**Comment by SBird1337** [comment]
@huderlem thats right I did not take that into account, if it is really 3 layers it might be worth considering to read the layers from common GFX Editors like Gimp or Photoshop (Though that should by no means be the priority when dealing with this issue...)

**Comment by garakmon** [comment]
Are there plans to add support for editing animated tiles?

**Comment by Delta456** [comment]
I think let's done with the general features and then we could move on with more features.

**Comment by SBird1337** [comment]
Thing is, animations are habdled by hand-crafted routines, idk if that should be for the map editor to handle
